packages/core/src/compiler/Style.ts

```diff
@@ -3116,6 +3113,15 @@ const translateExpr = (
       if (args.isErr()) {
         return addDiags(args.error, trans);
       }
+      const argsWithSourceLoc: ArgValWithSourceLoc<ad.Num>[] = [];
+      for (let i = 0; i < args.value.length; i++) {
+        const { start, end } = argExprs[i];
+        argsWithSourceLoc.push({
+          ...args.value[i],
+          start,
+          end,
+        });
+      }
```

Comment:
I see this in a few places; would it make sense to extract into a helper function?